src/ray/common/asio/instrumented_io_context.cc

```diff
@@ -18,7 +18,15 @@
 #include <iomanip>
 #include <iostream>
 #include <utility>
-
+#include "ray/stats/metric.h"
+
+DEFINE_stats(io_pool_req_num, "IO pool request num", {"Method"}, {}, ray::stats::GAUGE);
+DEFINE_stats(io_pool_req_execution_time_us, "IO pool execution time", {"Method"}, {},
+             ray::stats::GAUGE);
+DEFINE_stats(io_pool_req_queue_time_us, "IO pool queue time", {"Method"}, {},
```

Comment:
Is it worth further sub-categorizing these events? E.g.
"Type": "rpc call", "rpc handler", "timer", "callback", etc.
"Name": name

Reply:
I think to do that, we probably need to decouple this with asio context. Right now it's not passed. I can add RPC specific metrics to RPC module.